[Pinran] iP

Pinran-J opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen [(Source)](https://dansilvestre.com/productivity-quotes/)

 Duke helps you keep track of things to do, it's
- Text-based
- Easy to learn, easy to use
- ~~FAST~~ SUPER FAST

All you need to do is,
1. Download it from [here.](https://github.com/Pinran-J/ip/releases)
2. Double-click it
3. Add your tasks
4. Let it manage your task for you :kissing_heart:

And it's ***FREE***
<sup>for now</sub>

Features : 
- [x] Managing tasks
- [x] JavaFx GUI
- [ ] Secret features

## 
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```
 public static void main(String[] args) {
        Duke d = new Duke("data/", "save.txt");

        d.run();
    }
```

Commit:
Add Gradle support

Pinran-J committed:
Level 1. Greet, Echo, Exit

Pinran-J committed:
Level 2. Add, List

Pinran-J committed:
Level 3. Mark as Done

Pinran-J committed:
Level 4. ToDos, Events, Deadlines

Pinran-J committed:
Automated Text UI Testing

Pinran-J committed:
Level 5. Handle Errors

Pinran-J committed:
Level-6: Delete

Pinran-J committed:
Level 7. Save

Pinran-J committed:
Level 8. Dates and Times

Pinran-J committed:
Merge branch 'branch-Level-7'

Pinran-J committed:
Merge branch 'branch-Level-8'

Pinran-J committed:
Use More OOP

Pinran-J committed:
Organize into Packages

Pinran-J committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

Pinran-J committed:
Add JUnit Tests

Pinran-J committed:
Change mainClassName in build.gradle

Pinran-J committed:
Added JavaDoc

Pinran-J committed:
Complied with coding standard.

Pinran-J committed:
Added Find.

Pinran-J committed:
Merge branch 'branch-A-JavaDoc'

Pinran-J committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/EmptyDescriptionException.java
#	src/main/java/duke/Parser.java

Pinran-J committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

Pinran-J committed:
Resolved additional merge conflict

ling-ui commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,129 @@
+    private Storage storage;
+    private TaskList tList;
+    private Ui ui;
+    private Boolean active;
```
You can try naming your boolean isActive instead, to comply by the coding standards 😃

Pinran-J replied:
Fixed.

ling-ui commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,129 @@
+    /**
+     * Constructor for a Duke object.
+     *
+     * @param dirPath path of directory that storage use to locate save files.
```
For Javadoc, it is good practice to begin your sentence with a capital letter, in this case should be "Path of directory..." 😃

Pinran-J replied:
Fixed.

ling-ui commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,129 @@
+    }
+
+    /**
+     * Method that runs the Duke Chat bot.
```
Trivial issue, but you don't have to capitalise Chat and you can spell chatbot as one word. Might seem more professional to have consistent Javadoc. 😄

ling-ui commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,49 @@
+    private LocalTime end_time;
+
+    /**
+     * Constructor for an Event task
```
Remember to add a full stop at the end of your method summary 😃

Pinran-J replied:
Fixed

ling-ui commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,49 @@
+public class Event extends Task {
+
+    private String at;
+    private LocalDate start_date;
```
Variable names must be in camelCase. (i.e. startDate) This applies to the variables on the next few lines as well. 😃

ling-ui commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,111 @@
+     * @throws EmptyDescriptionException Throw if task description is empty.
+     * @throws InvalidFormatException Thrown if invalid task format.
+     */
+    public static String[] parseInput(String s) throws InvalidTaskException,
```
You can consider naming your method argument as something clearer, such as "input".

ling-ui commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,68 @@
+
+    /**
+     * Outputs the correct string to be saved in save file.
+     * @return String of data to be stored.
```
Remember to add a line spacing in your javadoc

Pinran-J replied:
Fixed

ling-ui commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,141 @@
+    /**
+     * Method that prints to screen a line for ease of reading.
+     */
+    public void line() {
```
For consistency you can consider naming this method as printLine() instead. 😄

Pinran-J replied:
Fixed.

ling-ui reviewed (commented):
Besides minor formatting issues, Javadoc was well written.

Pinran-J committed:
Use CheckStyle

Pinran-J committed:
Merge branch 'A-CheckStyle'

Pinran-J committed:
JavaFX tutorial part 3

Pinran-J committed:
Added Level-10

Pinran-J committed:
Fixed coding standard (Peer review comments)

WR3nd3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,124 @@
+     * @param s String that the user inputs.
+     * @return The string to be displayed by bot.
+     */
+    public String getResponse(String s) {
```
The control-flow of the method is very neat and easy to understand. Nice!

WR3nd3 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,114 @@
+     * @throws EmptyDescriptionException Throw if task description is empty.
+     * @throws InvalidFormatException Thrown if invalid task format.
+     */
+    public static String[] parseInput(String s) throws InvalidTaskException,
```
Despite its relative length, you managed to avoid deep nesting of if-else statements and were able to structure the code neatly. Well done!

WR3nd3 reviewed (commented):
LGTM to me, code is clear and well organized. Well done 👍🏻

Pinran-J committed:
Added colour to dialog boxes and circular display image

Pinran-J committed:
Add assert statements

Duke and Storage class : Absence of assert statements

Assert statements are important to ensure certain assumptions are held.

Assert statements have been added to Duke class to ensure proper parser
output and saving of files. Assert statements added to Storage to ensure
scanner has scanned the correct input.

Assertions are lightweight and can be used to check for bugs during
runtime.

Pinran-J committed:
Improve Parser code quality

Parser parseInput method: shorten method length

The method is very long with overlapping statements which increases the
method length further.

Long methods can affect the code’s readability.

Helper methods are being used to abstract the overlapping statements
which reduces the overall length of the method.

Pinran-J committed:
Improve Duke class code quality

Duke class getResponse method is too long with different levels of
abstraction within the code fragment.

The long method with different levels of abstract affects the overall
code quality.

Single Level of Abstraction Principle (SLAP) is being used to ensure
same level of abstraction within code fragment and also reduces
length of method.

Pinran-J committed:
Merge pull request #2 from Pinran-J/branch-A-Assertions

Add assert statements

Pinran-J committed:
Merge branch 'master' of https://github.com/Pinran-J/ip

Pinran-J committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java

Pinran-J committed:
Merge pull request #3 from Pinran-J/branch-A-CodeQuality

Branch A-CodeQuality

Pinran-J committed:
Merge branch 'master' of https://github.com/Pinran-J/ip

Pinran-J committed:
Updated build.gradle and fixed array size in parser

Pinran-J committed:
Changed duke image

Pinran-J committed:
Tweaked Duke's personality.

Pinran-J committed:
Updated product screenshot.

Pinran-J committed:
Update README.md

Pinran-J committed:
Update README.md

Pinran-J committed:
Update README.md

Pinran-J committed:
Update README.md

Pinran-J committed:
Update README.md
